processor/src/lib.rs

```diff
@@ -264,7 +273,12 @@ impl Process {
     /// - If the number of groups in a batch is not a power of 2, NOOPs are executed (one per
     ///   group) to bring it up to the next power of two (e.g., 3 -> 4, 5 -> 8).
     #[inline(always)]
-    fn execute_op_batch(&mut self, batch: &OpBatch) -> Result<(), ExecutionError> {
+    fn execute_op_batch(
+        &mut self,
+        batch: &OpBatch,
+        decorator_iter: &mut DecoratorIterator,
+        offset: usize,
```

Comment:
nit: let's rename `decorator_iter` to just `decorators` and `offset` to `op_offset`.

Reply:
Done